lib/engine/game/base.rb

```diff
@@ -650,6 +649,23 @@ def tile_cost(tile, entity)
         end
       end
 
+      def all_players
+        @players + @bankrupt_players
+      end
+
+      def declare_bankrupt(player)
+        if @bankrupt_players.include?(player)
+          game_error("#{player.name} is already bankrupt, cannot declare bankruptcy again.")
+        end
+
+        @bankrupt_players << player
+        @players.delete(player)
+      end
+
+      def game_error(msg)
```

Comment:
you don't use this method anywhere else, lets just do raise gmaeERror

Reply:
f95801cf